packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -196,7 +196,22 @@ class SliverConstraints extends Constraints {
   /// the earliest visible part of this sliver in the [AxisDirection].
   ///
   /// For example, if [AxisDirection] is [AxisDirection.down], then this is the
-  /// scroll offset at the top of the visible portion of the sliver.
+  /// amount the top of the visible portion of the sliver has been scrolled
+  /// past the top of the viewport.
```

Comment:
The top of the visible portion of the sliver, by definition, is at the top of the viewport or in the viewport.